sdk/tables/azure-data-tables/samples/async_samples/sample_encode_dataclass_model_async.py

```diff
@@ -0,0 +1,191 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_encode_dataclass_model_async.py
+
+DESCRIPTION:
+    These samples demonstrate the following: inserting entities into a table
+    and deleting entities from a table.
+
+USAGE:
+    python sample_encode_dataclass_model_async.py
+
+    Set the environment variables with your own values before running the sample:
+    1) TABLES_COSMOS_ENDPOINT_SUFFIX - the Table cosmos DB service account URL suffix
+    2) TABLES_COSMOS_ACCOUNT_NAME - the name of the cosmos DB account
+    3) TABLES_PRIMARY_COSMOS_ACCOUNT_KEY - the cosmos DB account access key
+"""
+import os
+import asyncio
+from datetime import datetime, timezone
+from uuid import uuid4, UUID
+from enum import Enum
+from dotenv import find_dotenv, load_dotenv
+from dataclasses import dataclass, asdict
+from typing import Optional
+from azure.data.tables import UpdateMode, EdmType
+from azure.data.tables.aio import TableClient
+
+
+@dataclass
+class Car:
+    PartitionKey: str
+    RowKey: str
+    price: Optional[float] = None
+    last_updated: Optional[datetime] = None
+    product_id: Optional[UUID] = None
+    inventory_count: Optional[int] = None
+    barcode: Optional[bytes] = None
+    color: Optional[Enum] = None
+    maker: Optional[str] = None
+    model: Optional[str] = None
+    production_date: Optional[datetime] = None
+    big_number: Optional[int] = None
+    is_second_hand: Optional[bool] = None
+
+
+class Color(Enum):
+    WHITE = "white"
+    GRAY = "gray"
+    BLACK = "black"
+
+
+def encode_int(value):
+    return EdmType.STRING, str(value)
+
+def encode_uuid(value):
+    return None, str(value)
+
+encoder_map = {
+    int: encode_int,
+    UUID: encode_uuid,
+}
+
+def decode_string(value):
+    try:
+        return int(value)
```

Comment:
Which property was this added to support?

Reply:
Instead of this, I would probably use something like:
```python
def decode_string(value):
    try:
        return Color(value)  # deserialize the enum value
    except ValueError:
        return value
```

Reply:
When encoding, big int stored as string in DB. So, we should convert it back when decoding. I think we can have both - deserializing enum and int - in this custom converter.